Frameworks/CoreGraphics/DWriteWrapper.mm

```diff
@@ -479,4 +479,170 @@ HRESULT _DWriteFontGetBoundingBoxesForGlyphs(
     });
 
     return ret;
-}
\ No newline at end of file
+}
+
+#pragma region CGFontCreateWithDataProvider helpers
+
+/**
+ * Custom implementation of IDWriteFontFileStream that implements Read in terms of an underlying CFDataRef
+ */
+class DWriteFontBinaryDataStream : public RuntimeClass<RuntimeClassFlags<WinRtClassicComMix>, IDWriteFontFileStream> {
+protected:
+    InspectableClass(L"Windows.Bridge.DirectWrite", TrustLevel::BaseTrust);
+
+public:
+    DWriteFontBinaryDataStream() {
+    }
+
+    HRESULT RuntimeClassInitialize(CGDataProviderRef dataProvider) {
+        _data.reset(CGDataProviderCopyData(dataProvider));
+
+        // Just use current time for _lastWriteTime
+        FILETIME fileTime;
+        SYSTEMTIME systemTime;
+
+        GetSystemTime(&systemTime);
+        // Returns 0 on failure, which is probably good enough for WinObjC purposes?
+        SystemTimeToFileTime(&systemTime, &fileTime);
```

Comment:
>SystemTimeToFileTime [](start = 8, length = 20)

also preferably, if this fails, you can just set the time to 0 and leave, rather than running through the below code.